framework/src/transfers/MultiAppUserObjectTransfer.C

```diff
@@ -49,20 +47,20 @@ validParams<MultiAppUserObjectTransfer>()
 }
 
 MultiAppUserObjectTransfer::MultiAppUserObjectTransfer(const InputParameters & parameters)
-  : MultiAppTransfer(parameters),
-    _to_var_name(getParam<AuxVariableName>("variable")),
+  : MultiAppFieldTransfer(parameters),
     _user_object_name(getParam<UserObjectName>("user_object")),
     _all_master_nodes_contained_in_sub_app(getParam<bool>("all_master_nodes_contained_in_sub_app"))
 {
   // This transfer does not work with DistributedMesh
   _fe_problem.mesh().errorIfDistributedMesh("MultiAppUserObjectTransfer");
-}
 
-void
-MultiAppUserObjectTransfer::initialSetup()
-{
-  if (_direction == TO_MULTIAPP)
-    variableIntegrityCheck(_to_var_name);
+  if (_to_var_names.size() != 1)
+    paramError("variable", " Support single to-variable only ");
+
+  if (_from_var_names.size() > 0)
+    paramError("source_variable",
+               " You should not provide any source variables since the transfer takes values from "
+               "user objects ");
```

Comment:
Oh - so for a derived class you don't need to provide source_variables. Perhaps a better design would be to use the `suppressParameter()` method in InputParameters. This will allow you to leave it as required in the base but effectively delete it in this once class.

Reply:
A really cool suggestion!